enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/discovery/HazelcastClusterTopology.java

```diff
@@ -225,9 +226,11 @@ static void casLeaders( HazelcastInstance hazelcastInstance, LeaderInfo leaderIn
 
         boolean greaterTermExists = termComparison > 0;
 
-        boolean invalidTerm = greaterTermExists || ( termComparison == 0 && !leaderInfo.isSteppingDown() );
+        boolean sameTermButNoStepdown = termComparison == 0 && !leaderInfo.isSteppingDown();
 
-        boolean success = !( invalidTerm || noUpdate);
+        boolean invalidTerm = greaterTermExists || sameTermButNoStepdown;
+
+        boolean success = !( invalidTerm || noUpdate );
 
         if ( !success )
```

Comment:
I think this if-statement would be clearer if you just wrote all the conditions verbatim:

```
if ( noUpdate || greaterTermExists || sameTermButNoStepDown )
{
    return;
}
```

This removes the slightly ambiguous `success` and `invalidTerm`.

Perhaps noUpdate might also be renamed to sameLeader?